File: app/lib/models/job_model.dart
class JobData {
  final String id;
  final String title;
  final String description;
  final String type; // "vacancy" or "seeker"
  final String role;
  final String city;
  final String state;
  final String contactName;
  final String contactPhone;
  final String? contactEmail;
  final String? link;
  final DateTime createdAt;

  JobData({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.role,
    required this.city,
    required this.state,
    required this.contactName,
    required this.contactPhone,
    this.contactEmail,
    this.link,
    required this.createdAt,
  });

  factory JobData.fromJson(Map<String, dynamic> json) {
    return JobData(
      id: (json['_id'] ?? json['id'] ?? '') as String,
      title: json['title'] as String? ?? '',
      description: json['description'] as String? ?? '',
      type: json['type'] as String? ?? 'vacancy',
      role: json['role'] as String? ?? '',
      city: json['city'] as String? ?? '',
      state: json['state'] as String? ?? '',
      contactName: json['contactName'] as String? ?? '',
      contactPhone: json['contactPhone'] as String? ?? '',
      contactEmail: json['contactEmail'] as String?,
      link: json['link'] as String?,
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}

File: backend/src/features/jobs/job.model.ts
import mongoose from "mongoose";

const applicationSchema = new mongoose.Schema(
  {
    applicantId: {
      type: String,
      required: true,
    },
    message: {
      type: String,
    },
    status: {
      type: String,
      enum: ["pending", "approved", "rejected"],
      default: "pending",
    },
  },
  {
    _id: false,
    timestamps: { createdAt: true, updatedAt: false },
  },
);

const jobSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true,
    },
    description: {
      type: String,
      required: true,
      trim: true,
    },
    type: {
      type: String,
      enum: ["vacancy", "seeker"],
      required: true,
    },
    role: {
      type: String,
      required: true,
      trim: true,
    },
    city: {
      type: String,
      required: true,
      trim: true,
    },
    state: {
      type: String,
      required: true,
      trim: true,
    },
    contactName: {
      type: String,
      required: true,
      trim: true,
    },
    contactPhone: {
      type: String,
      required: true,
      trim: true,
    },
    contactEmail: {
      type: String,
      trim: true,
    },
    link: {
      type: String,
      trim: true,
    },
    createdBy: {
      type: String,
      required: true,
      index: true,
    },
    status: {
      type: String,
      enum: ["pending", "approved", "rejected"],
      default: "approved",
    },
    applications: [applicationSchema],
  },
  {
    timestamps: true,
  },
);

const Job = mongoose.model("Job", jobSchema);

export default Job;

